### DISCUSSION TITLE
How to avoid double data changes while requesting paginated data?

### DISCUSSION THREAD

**Discussion by wqbill** [TanStack/query#6713] [Q&A]
Here is the thing:
Page1: A B
Page2: C D
but Page1 changed to A C while I went to page 1 again.
**C D->A B(cached data)>A C**
The rendered list will change twice.
```
{
  cacheTime: 0,
  keepPreviousData: true
}
```
I'm using this kind of options for now.Is this the best practice for this situation?

**Comment by TkDodo** [comment]
that's the concept of `stale-while-revalidate`. We show stale data (`AB`) while getting new data `AC` and then showing that. The drawback of having your list go from `AB` to `AC` occasionally is better than always showing a loading spinner. Suppose the data didn't change, then users will see a spinner every time they change pages. That's what they'll get with `cacheTime: 0`. So, it's a tradeoff, and you have to lean one way or the other - can't have the cake and eat it, too :)
